src/state/user/hooks/index.tsx

```diff
@@ -404,15 +406,26 @@ export function useRemoveUserAddedToken(): (chainId: number, address: string) =>
 
 export function useGasPrice(chainIdOverride?: number): string {
   const { chainId: chainId_, chain } = useActiveWeb3React()
+  const library = useWeb3LibraryContext()
   const chainId = chainIdOverride ?? chainId_
-  const userGas = useSelector<AppState, AppState['user']['gasPrice']>((state) => state.user.gasPrice)
+  const { data: bscProviderGasPrice = GAS_PRICE_GWEI.default } = useSWR(
+    library && library.provider && chainId === ChainId.BSC && ['bscProviderGasPrice', library.provider],
+    async () => {
+      const gasPrice = await library.getGasPrice()
+      return gasPrice.toString()
+    },
+    {
+      revalidateOnFocus: false,
+      revalidateOnReconnect: false,
+    },
+  )
   const { data } = useFeeData({
     chainId,
     enabled: chainId !== ChainId.BSC && chainId !== ChainId.BSC_TESTNET,
```

Comment:
Can just remove this line and reuse the feedata?

Reply:
Suggest keeping this hook as a simple current gas market price hook. 

It shouldn't be used to send transactions